### PR TITLE
Normalize DS4 and DS6 icons

### LINKED ISSUE #472: Icon: normalize ds4/ds6 docs and icon names
Not much we can do about having different icon names in ds4 and ds6, so we are always going to require *some* custom documentation, but there is some duplication we can avoid such as the intros to background and foreground icons.

We should also try and identify which icons are common across ds4 and ds6 and standardize those using the ds6 naming convention and move those to a `common/icon.html` include.

Finally, let's clarify when to use aria-hidden (presentational icons) and when to use aria-label (non-presentational icons).

### PR DESCRIPTION
**Pull request by ianmcburnie**
# Closes #472 

## Goal(s)

1. Normalize icon docs
1. Identify icons that are common across ds4 and ds6 (renaming aliases where appropriate)
1. Identify icons that are unique to ds4 and ds6

## Main changes

1. Created a `common/icon.html` file and `ds4/icon-custom.html` and `ds6/icon-custom.html`
1. Reformatted background and foreground icon sub sections
1. Added accessibility info for icons
1. Updated DS4 icon examples to use the DS6 "demo-tile" style
1. Removed all occurrences of background icons (bar one) from the website and removed skin-base64.min.css stylesheet. This improves performance and we should really be focussing efforts on foreground SVG anyhow.
1. Removed SVG source code from SVG module section. Added a link to the SVG source on GitHub instead. Maintaining the duplicate symbols under `docs` and `src/svg` was not fun.

### New DS4 aliases

* information -> info
* view-grid -> filter-grid
* view-detail - > filter-single
* view-list -> filter-list
* check -> tick
* mail -> messages
* mail-open -> messages-open
* notifications -> notification
* notifications-null -> notification-null
* save -> download
* follow -> add
* sell -> selling
* sell-null -> selling-null

### New DS6 aliases

* filter-grid -> filter-gallery (breaking change, I removed filter-grid).

### Notes

For DS4, I've left in the old aliases for backwards compatibility, but given the volume of them I'm starting to think they should be removed. Thoughts?

All aliases taken from Icon Library DS6.v1.01.

I notice that some DS6 icons have been removed from the playbook. I've removed these from our site for now too:

* sign-out
* user-profile
* flag

### Tech Debt

* We do not have a DS4 foreground icon for `icon-selling`. I've hardcoded the website to use the background icon for now. 
* I added some inline styling to the icon section. Given that our index.less is a mess, and we are planning on creating a new website soon, I'm okay with leaving this for now.
* We need to add a test page containing all of the ds6 background icons.
* Added the DS4 "open in new window" icon to DS6. We need this icon for a11y, and can update it when we get a new design

## Icon groups identical across DS4 and DS6

screen shot 2019-01-22 at 2 54 19 pm

screen shot 2019-01-22 at 2 55 18 pm

## Icon groups with some common icons across DS4 and DS6

screen shot 2019-01-22 at 2 57 59 pm

screen shot 2019-01-22 at 2 57 48 pm

## Icon Groups unique to DS4

screen shot 2019-01-22 at 2 58 58 pm

screen shot 2019-01-22 at 2 59 04 pm

## Icon Groups unique to DS6

screen shot 2019-01-22 at 2 59 12 pm

### CODE REVIEW

File: src/svg/ds4/icons.svg [hunk 1]
```diff
@@ -273,15 +302,15 @@
...
+    </symbol>
+    <symbol id="icon-filter-single" viewBox="0 0 22 23">
         <path d="M21.824 13.591a.599.599 0 0 1-.597.597H.597A.599.599 0 0 1 0 13.591V.597C0 .269.268 0 .597 0h20.63a.6.6 0 0 1 .597.597v12.994zm.069 8.212a.6.6 0 0 1-.597.597H.684a.6.6 0 0 1-.597-.597v-4.352c0-.328.268-.597.597-.597h20.613a.6.6 0 0 1 .597.597v4.352h-.001z"/>
     </symbol>
-    <symbol id="icon-view-grid" viewBox="0 0 22 22">
+    <symbol id="icon-filter-gallery" viewBox="0 0 22 22">
```
Comment (seangates) [inline src/svg/ds4/icons.svg]: Please see my other comment on changing these IDs.

File: src/svg/ds4/icons.svg [hunk 2]
```diff
@@ -273,15 +302,15 @@
     <symbol id="icon-underline" viewBox="0 0 21 24">
         <path d="M16.372 18.345a8.422 8.422 0 0 1-5.94 1.901 9.348 9.348 0 0 1-6.137-1.834 7.052 7.052 0 0 1-2.185-5.787L2.108.001h2.709v12.651a5.28 5.28 0 0 0 1.458 4.03 5.828 5.828 0 0 0 4.2 1.368 5.278 5.278 0 0 0 3.929-1.374 5.492 5.492 0 0 0 1.349-4.043L15.754 0h2.709v12.651a7.315 7.315 0 0 1-2.092 5.692l.001.002zM0 22.099h20.571v1.903H0v-1.903z"/>
     </symbol>
-    <symbol id="icon-view-detail" viewBox="0 0 22 23">
+    <symbol id="icon-filter-list" viewBox="0 0 23 23">
```
Comment (seangates) [inline src/svg/ds4/icons.svg]: Please see my other comment on changing these IDs.

File: src/svg/ds6/icons.svg [hunk 3]
```diff
@@ -129,7 +129,7 @@
     <symbol id="icon-fast-n-free" viewBox="1.33 4 21.33 16">
         <path d="M19.758 8h-2.91V4H3.273c-1.067 0-1.94.9-1.94 2v11h1.94c0 1.66 1.3 3 2.909 3 1.61 0 2.909-1.34 2.909-3h5.818c0 1.66 1.3 3 2.91 3 1.609 0 2.908-1.34 2.908-3h1.94v-5l-2.91-4zM6.182 18.5c-.805 0-1.455-.67-1.455-1.5s.65-1.5 1.455-1.5 1.454.67 1.454 1.5-.65 1.5-1.454 1.5zm13.09-9l1.901 2.5h-4.325V9.5h2.425zm-1.454 9c-.805 0-1.454-.67-1.454-1.5s.65-1.5 1.454-1.5c.805 0 1.455.67 1.455 1.5s-.65 1.5-1.455 1.5z"/>
     </symbol>
-    <symbol id="icon-filter-grid" viewBox="0.25 0.75 19.5 16.5">
+    <symbol id="icon-filter-gallery" viewBox="0.25 0.75 19.5 16.5">
```
[the remark this answers is not in the document]
Comment (ianmcburnie) [reply src/svg/ds6/icons.svg]: These names come from the playbook. I'm trying to be consistent with that.

screen shot 2019-01-22 at 10 14 03 pm

[the remark this answers is not in the document]
Comment (seangates) [reply src/svg/ds6/icons.svg]: Gotcha!

I could have _sworn_ I pulled mine from the playbook, too, as I was creating the icons. Maybe their file names didn't match up to their playbook?

File: dist/less/ds4/mixins.less [hunk 4]
```diff
@@ -603,14 +648,29 @@
...
+
+.icon-filter-list(@width: @ds4-icon-filter-list-width; @height: @ds4-icon-filter-list-height) {
+    .background-icon-image(@width; @height; @ds4-icon-filter-list-base64);
+}
+
+.icon-filter-single(@width: @ds4-icon-filter-single-width; @height: @ds4-icon-filter-single-height) {
+    .background-icon-image(@width; @height; @ds4-icon-filter-single-base64);
+}
+
+// deprecated in favour of filter-grid
```
Comment (seangates) [inline dist/less/ds4/mixins.less]: Should be `filter-gallery`

Comment (seangates) [reply dist/less/ds4/mixins.less]: I didn't see this fixed, but if you're going to remove these deprecated ones then it won't even matter.

Comment (ianmcburnie) [reply dist/less/ds4/mixins.less]: I changed it in the src. Maybe I forgot to build the dist file...

**Comment by seangates** [comment]
@ianmcburnie Looks like you need to rebuild your `dist` files:
image

**Comment by ianmcburnie** [comment]
Done.

**Pull request merged**